pkg/autoscaler/autoscaler.go

```diff
@@ -429,15 +436,62 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
 			return true
 		}, framework.WaitMedium, pollingInterval).Should(BeZero())
 
+		// Wait until the MachineSets have reach minimum size so that
+		// we can test to ensure that the remaining Machines and Nodes
+		// do not contain deletion information.
+		By("Waiting for transient machinesets to reach minimum size")
+		transientMachineSetSelector := &metav1.LabelSelector{
+			MatchLabels: map[string]string{machinesetTestLabel: ""},
+		}
+		Eventually(func() bool {
+			transientMachineSets, err := framework.GetMachineSets(client, transientMachineSetSelector)
+			Expect(err).NotTo(HaveOccurred())
+			for _, tms := range transientMachineSets {
+				if tms.Status.Replicas != transientMachineSetsMinSize {
+					glog.Infof("Waiting for all transient machinesets to reach minimum size")
+					return false
+				}
+			}
+			glog.Infof("All transient machinesets have reached minimum size")
+			return true
+		}, framework.WaitLong, pollingInterval).Should(BeTrue())
+
+		By("Checking to ensure that no deletion annotations exist on machines")
+		transientMachineSets, err := framework.GetMachineSets(client, transientMachineSetSelector)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(func() bool {
+			annotationsfound := 0
+			for _, tms := range transientMachineSets {
+				machines, err := framework.GetMachinesFromMachineSet(client, &tms)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
Similar to above, could return the error to get retries

Reply:
++